pmoves/services/hi-rag-gateway-v2/rerank.py

```
@@ -0,0 +1,76 @@
+    import config as _cfg
+    if _cfg._reranker is not None:
+        return _cfg._reranker
+    if not RERANK_ENABLE:
+        return None
+    try:
+        use_fp16 = _RERANK_FP16_EFFECTIVE
+        _cfg._reranker = FlagReranker(RERANK_MODEL_RESOLVED, use_fp16=use_fp16)
+        if DEVICE == "cuda" and hasattr(_cfg._reranker, "model"):
+            _cfg._reranker.model = _cfg._reranker.model.to("cuda")  # type: ignore[attr-defined]
+        return _cfg._reranker
+    except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)